scripts/db_migrator.py

```diff
@@ -596,6 +596,58 @@ def migrate_route_table(self):
                     route_key = "ROUTE_TABLE:{}".format(route_prefix)
                 self.appDB.set(self.appDB.APPL_DB, route_key, 'weight','')
 
+
+    def update_edgezone_aggregator_config(self):
+        """
+        Update cable length configuration in ConfigDB for T0 neighbor interfaces
+        connected to EdgeZone Aggregator devices, while resetting the port values to trigger a buffer change
+        1. Find T1 cable length - this will be the cable length to the EdgeZone Aggregator device.
+        2. Find a list of all interfaces connected to an EdgeZone Aggregator device.
+        3. Update CABLE_LENGTH values for these interfaces with the length found in (1).
+        4. Reset port values to trigger buffer manager event listener to bring across new buffer changes for specified ports
+        """
+        device_neighbor_metadata = self.configDB.get_table("DEVICE_NEIGHBOR_METADATA")
+        device_neighbors = self.configDB.get_table("DEVICE_NEIGHBOR")
+        cable_length = self.configDB.get_table("CABLE_LENGTH")
+        port_table = self.configDB.get_table("PORT")
+        edgezone_aggregator_devs = []
+        edgezone_aggregator_intfs = []
+        EDGEZONE_AGG_CABLE_LENGTH = "5m"
+        for k, v in device_neighbor_metadata.items():
+            if "type" in v:
+                if v["type"] == "LeafRouter":
+                    T1_device = k
+                elif v["type"] == "EdgeZoneAggregator":
+                    edgezone_aggregator_devs.append(k)
+
+        if len(edgezone_aggregator_devs) == 0:
+            return
+
+        for intf, intf_info in device_neighbors.items():
+            if intf_info["name"] == T1_device:
+                T1_intf = intf
+            elif intf_info["name"] in edgezone_aggregator_devs:
+                edgezone_aggregator_intfs.append(intf)
+
+
+        if cable_length["AZURE"][T1_intf] is not None:
+            EDGEZONE_AGG_CABLE_LENGTH = cable_length["AZURE"][T1_intf]
+
+        cable_length_table = self.configDB.get_entry("CABLE_LENGTH", "AZURE")
+
+        for intf, length in cable_length_table.items():
+            if intf in edgezone_aggregator_intfs:
+                cable_length_table[intf] = EDGEZONE_AGG_CABLE_LENGTH
+
+        """ Set new cable length values """
+        self.configDB.set_entry("CABLE_LENGTH", "AZURE", cable_length_table)
+
+        for intf, intf_items in port_table.items():
+            if intf in edgezone_aggregator_intfs:
+                """ Reset interface values to current value so that buffer manager listener picks up new cable length change """
+                self.configDB.set_entry("PORT", intf, intf_items)
```

Comment:
Please confirm in your testing if this is really needed. If it is needed, would prefer to just change the speed alone

Reply:
Checking

Reply:
This is needed - since the get_entry function returns the whole sub-dict (a dict inside a dict), resetting its value vs just the speed has the same effect.

Reply:
Can you check if setting the speed is necessary? I would prefer to avoid setting anything in the port table